third_party/window/window.go

```diff
@@ -0,0 +1,87 @@
+// source: https://github.com/jbrukh/window
+package store
+
+// An array-based moving window; a moving window
+// is a queue with a maximum size and the property
+// that when the size is reached, pushing a new
+// element into the queue causes the head to be
+// popped.
+//
+// You can optimize the amount of slice copying
+// that the MovingWindow will be doing by trading
+// off with space complexity. Namely, the underlying
+// array is allocated with a size that is the
+// multiple of the intended capacity of the queue
+// so that copying is less frequent.
+type MovingWindow struct {
+	arr  []interface{}
```

Comment:
Building on that - this code is rather simple, we could probably reimplement it for our needs under our own license in the `store/daystore` package with the usage of `[]hourEntry` instead of `[]interface{}`.